src/main/java/com/amazon/opendistroforelasticsearch/sql/metrics/Counter.java

```diff
@@ -0,0 +1,27 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.metrics;
+
+public interface Counter<T> {
+
+    void increment();
+
+    void increment(long n);
```

Comment:
Should we stick to java's naming convention (see `LongAdder::add(long)`) and call this method `add` instead?

Reply:
Sure, will change it.